finetuner/tuner/metrics.py

```diff
@@ -0,0 +1,164 @@
+        user as matching ground truth, should be under ``doc.tags[finetuner.__default_tag_key__]['targets']``.
+    :param k: the point at which evaluation is computed, if ``None`` is given the entire outputs sequence will
+        be considered.
+    :return: the number of hits.
+    """
+    relevance, _ = _doc_to_relevance(doc)
+    return hit_at_k(binary_relevance=relevance, k=k)
+
+
+def reciprocal_rank(doc: Document) -> float:
+    """
+    Compute the reciprocal rank metric for a Jina document
+    :param doc: A Jina document. The matched document identifiers, predicted by the search system, are expected to
+        be under ``doc.matches``. The expected documents identifiers along with the relevance scores, given by the
+        user as matching ground truth, should be under ``doc.tags[finetuner.__default_tag_key__]['targets']``.
+    :return: Reciprocal rank score.
+    """
+    relevance, _ = _doc_to_relevance(doc)
+    return _reciprocal_rank(binary_relevance=relevance)
+
+
+def dcg(doc: Document, k: Optional[int] = None) -> float:
+    """ "
+    Compute the discounted cumulative gain (DCG score) for a Jina document
+    :param doc: A Jina document. The matched document identifiers, predicted by the search system, are expected to
+        be under ``doc.matches``. The expected documents identifiers along with the relevance scores, given by the
+        user as matching ground truth, should be under ``doc.tags[finetuner.__default_tag_key__]['targets']``.
+    :param k: The number of documents in each of the lists to consider in the DCG computation. If ``None`` is given,
+        the complete lists are considered for the evaluation.
+    :return: The NDCG score.
+    """
+    relevance, _ = _doc_to_relevance(doc)
+    return dcg_at_k(relevance=relevance, k=k)
+
+
+def ndcg(doc: Document, k: Optional[int] = None) -> float:
+    """ "
+    Compute the normalized discounted cumulative gain (NDCG score) for a Jina document
+    :param doc: A Jina document. The matched document identifiers, predicted by the search system, are expected to
+        be under ``doc.matches``. The expected documents identifiers along with the relevance scores, given by the
+        user as matching ground truth, should be under ``doc.tags[finetuner.__default_tag_key__]['targets']``.
+    :param k: The number of documents in each of the lists to consider in the NDCG computation. If ``None`` is
+        given, the complete lists are considered for the evaluation.
+    :return: The NDCG score.
+    """
+    relevance, _ = _doc_to_relevance(doc)
+    return ndcg_at_k(relevance=relevance, k=k)
+
+
+METRICS = {
+    "r_precision": r_precision,
+    "precision": precision,
+    "recall": recall,
+    "f1score": f1score,
+    "average_precision": average_precision,
+    "hit": hit,
+    "reciprocal_rank": reciprocal_rank,
+    "dcg": dcg,
+    "ndcg": ndcg,
+}
```

Comment:
i have one suggestion, use `metrics` as the classmethod of `Evaluator`, such as 

```python
class Evaluator:

    @classmethod
    def list_available_metrics(cls):
        return ...
```

when user don't know what to use as evaluation function, this might provide some hints.

Reply:
good point, added!